### PR TITLE
Don't generate proxy classes for embedded documents

### LINKED ISSUE #1627: Don't generate proxy classes for embedded documents
Originally reported as https://github.com/doctrine/common/issues/813.

With doctrine/common 2.8.0, generating proxy objects for final classes throws an error. Since documents should not be final, this does not cause any issues. However, ODM also generates proxy classes for embedded documents, which shouldn't happen because they won't be used.

### PR DESCRIPTION
**Pull request by alcaeus**
Fixes #1627.

Not sure about tests here, but it fixes the issue.

### CODE REVIEW

File: lib/Doctrine/ODM/MongoDB/Tools/Console/Command/GenerateProxiesCommand.php [hunk 1]
```diff
@@ -62,7 +63,9 @@ protected function execute(Console\Input\InputInterface $input, Console\Output\O
     {
         $dm = $this->getHelper('documentManager')->getDocumentManager();
         
-        $metadatas = $dm->getMetadataFactory()->getAllMetadata();
+        $metadatas = array_filter($dm->getMetadataFactory()->getAllMetadata(), function (ClassMetadata $classMetadata) {
+            return !$classMetadata->isEmbeddedDocument;
```
Comment (malarzm) [inline lib/Doctrine/ODM/MongoDB/Tools/Console/Command/GenerateProxiesCommand.php]: I think we should also exclude `$classMetadata->isQueryResultDocument` while we're here (and maybe `$classMetadata->isMappedSuperclass` too)?

**Comment by malarzm** [comment]
> Not sure about tests here, but it fixes the issue.

I wouldn't worry too much here, it's only commands ;)

File: lib/Doctrine/ODM/MongoDB/Tools/Console/Command/GenerateProxiesCommand.php (hunk 1 above)
Comment (alcaeus) [inline lib/Doctrine/ODM/MongoDB/Tools/Console/Command/GenerateProxiesCommand.php]: I'll just change the check to `isDocument`, that's simpler. I'll have to think about `isQueryResultDocument` - there may be the need for proxies there.

Comment (alcaeus) [inline lib/Doctrine/ODM/MongoDB/Tools/Console/Command/GenerateProxiesCommand.php]: Updated. Turns out we don't have `isDocument`, so I can't test against that. We also don't need to generate proxies for query result documents, so I'll add that when merging down to master. 👍

**Pull request merged**